services/provider/server/server.go

```diff
@@ -373,33 +373,26 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 	})
 
 	if consumerResource.Spec.StorageQuotaInGiB > 0 {
-		clusterResourceQuotaSpec := &quotav1.ClusterResourceQuotaSpec{
-			Selector: quotav1.ClusterResourceQuotaSelector{
-				LabelSelector: &metav1.LabelSelector{
-					MatchExpressions: []metav1.LabelSelectorRequirement{
-						{
-							Key:      string(consumerResource.UID),
-							Operator: metav1.LabelSelectorOpDoesNotExist,
-						},
-					},
-				},
-			},
-			Quota: corev1.ResourceQuotaSpec{
-				Hard: corev1.ResourceList{"requests.storage": *resource.NewScaledQuantity(
-					int64(consumerResource.Spec.StorageQuotaInGiB),
-					resource.Giga,
-				)},
-			},
-		}
 
 		extR = append(extR, &pb.ExternalResource{
 			Name: "QuotaForConsumer",
 			Kind: "ClusterResourceQuota",
-			Data: mustMarshal(clusterResourceQuotaSpec),
+			Data: mustMarshal(s.getClusterResourceQuotaSpec(consumerResource)),
 		})
 
 	}
 
+	desiredClientConfigMap, err := s.getDesiredClientConfigMap(ctx, consumerResource)
+	if err != nil {
+		return nil, fmt.Errorf("failed to prepare desired config map %v", err)
+	}
+
+	extR = append(extR, &pb.ExternalResource{
+		Name: "DesiredClientConfig",
+		Kind: "ConfigMap",
+		Data: mustMarshal(desiredClientConfigMap),
+	})
```

Comment:
What is this new config map used for? If the intern is to send the hash this is the wrong mechanism. The has is a metadata of the entire operation, and should not be treated as another external resource.